projects/packages/my-jetpack/_inc/hooks/use-notification-watcher/use-expiring-soon-plans-notice.tsx

```diff
@@ -0,0 +1,79 @@
+import { __ } from '@wordpress/i18n';
+import { useContext, useEffect, useCallback } from 'react';
+import { NOTICE_PRIORITY_MEDIUM } from '../../context/constants';
+import { NoticeContext } from '../../context/notices/noticeContext';
+import useAnalytics from '../use-analytics';
+import { useGetExpiringSoonNoticeContent } from './use-get-expiring-soon-notice-content';
+import type { NoticeOptions } from '../../context/notices/types';
+
+type RedBubbleAlerts = Window[ 'myJetpackInitialState' ][ 'redBubbleAlerts' ];
+
+const useExpiringSoonPlansNotice = ( redBubbleAlerts: RedBubbleAlerts ) => {
```

Comment:
This hook and the `useExpiredPlansNotice` hook look extremely similar to me. Do you think it'd be possible to combine these two hooks into one with a few conditionals?

Reply:
The events could be named the same thing, just with a unique property that specifies which one was clicked

Reply:
Yes, will do. 👍